cmd/reflect/reflect.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2020 Authors of Hubble
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reflect
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/cilium/hubble/pkg/defaults"
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+	rpb "google.golang.org/grpc/reflection/grpc_reflection_v1alpha"
+)
+
+var (
+	serverURL string
+	output    string
+)
+
+// New reflect command.
+func New() *cobra.Command {
+	reflectCmd := &cobra.Command{
+		Use:   "reflect",
+		Short: "Let Hubble reflect upon itself",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runReflect()
+		},
+	}
+	reflectCmd.Flags().StringVarP(&serverURL, "server", "", defaults.GetDefaultSocketPath(), "URL to connect to server")
+	reflectCmd.Flags().StringVarP(&output, "output", "o", "compact", "\"compact\" or \"json\"")
+	return reflectCmd
+}
+
+func runReflect() (err error) {
+	conn, err := grpc.Dial(serverURL, grpc.WithInsecure())
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if closeErr := conn.Close(); closeErr != nil {
+			err = closeErr
```

Comment:
I think it would be better to assign `err` only if it's `nil` because you will otherwise overwrite the original error. We could also consider error wrapping.

Reply:
ah yes that makes sense. i just put some random stuff i found on stackoverflow, trying to figure out what the best practice is for dealing with errors returned from deferred functions.